src/main/java/com/blackducksoftware/integration/hub/alert/channel/DistributionChannel.java

```diff
@@ -22,19 +22,82 @@
  */
 package com.blackducksoftware.integration.hub.alert.channel;
 
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringEscapeUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.jpa.repository.JpaRepository;
+
 import com.blackducksoftware.integration.hub.alert.MessageReceiver;
 import com.blackducksoftware.integration.hub.alert.datasource.entity.DatabaseEntity;
+import com.blackducksoftware.integration.hub.alert.digest.model.CategoryData;
+import com.blackducksoftware.integration.hub.alert.digest.model.ItemData;
+import com.blackducksoftware.integration.hub.alert.digest.model.ProjectData;
 import com.blackducksoftware.integration.hub.alert.event.AbstractChannelEvent;
+import com.blackducksoftware.integration.hub.notification.processor.ItemTypeEnum;
+import com.blackducksoftware.integration.hub.notification.processor.NotificationCategoryEnum;
 import com.google.gson.Gson;
 
 public abstract class DistributionChannel<E extends AbstractChannelEvent, C extends DatabaseEntity> extends MessageReceiver<E> {
+    private final static Logger logger = LoggerFactory.getLogger(DistributionChannel.class);
+    protected final JpaRepository<C, Long> repository;
 
-    public DistributionChannel(final Gson gson, final Class<E> clazz) {
+    public DistributionChannel(final Gson gson, final JpaRepository<C, Long> repository, final Class<E> clazz) {
         super(gson, clazz);
+        this.repository = repository;
     }
 
     public abstract void sendMessage(final E event, final C config);
 
     public abstract String testMessage(final C config);
 
+    protected String createHtmlMessage(final ProjectData projectData) {
```

Comment:
I could be wrong but it looks like slack recommends json as the message body 
https://api.slack.com/web

Reply:
I take this message and insert it into json in my slack_channel. But I will now go and remove the HTML elements and make my own specific channel message.